app/searchApi.rb

```diff
@@ -210,16 +214,19 @@ def get_query_display(params)
 
 def aws_encode(params, facetTypes)
   aws_params = {
-    query: params['q'] ? params['q'].join(" ") : 'test',
+    query: params.has_key?('q') && params['q'].length > 0 ? params['q'].join(" ") : 'matchall',
     size: params['rows'].length > 0 ? params['rows'][0] : 10,
     sort: params['sort'].length > 0 ? SORT[params['sort'][0]] : '_score desc',
     start: params['start'].length > 0 ? params['start'][0] : 0,
   }
-  
+
+  if aws_params[:query] == 'matchall' then aws_params[:query_parser] = "structured" end
+
   # create facet query, only create facet query for fields specified in facetTypes
   facetQuery = {}
   facetTypes.each { |facetType| facetQuery[facetType] = FACETS[facetType]['awsFacetParam'] }
-  aws_params[:facet] = JSON.generate(facetQuery)
+  
+  if !facetQuery.empty? then aws_params[:facet] = JSON.generate(facetQuery) end
```

Comment:
Ruby evaluates the left-hand-side of an 'or' expression, and only if it's true does it evaluate the right-hand-side. So it's like saying "do this *or* do that, but not both", with the preference on the left.

In this case, either it's an empty query, or else you want to generate the `aws_params`.